pkg/controller/mongodb/replica_set_controller.go

```diff
@@ -78,18 +80,21 @@ func Add(mgr manager.Manager) error {
 type ManifestProvider func() (automationconfig.VersionManifest, error)
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager, manifestProvider ManifestProvider) reconcile.Reconciler {
+func newReconciler(mgr manager.Manager, manifestProvider ManifestProvider) *ReplicaSetReconciler {
 	mgrClient := mgr.GetClient()
+	secretWatcher := watch.New()
+
 	return &ReplicaSetReconciler{
 		client:           kubernetesClient.NewClient(mgrClient),
 		scheme:           mgr.GetScheme(),
 		manifestProvider: manifestProvider,
 		log:              zap.S(),
+		secretWatcher:    &secretWatcher,
 	}
 }
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
+func add(mgr manager.Manager, r *ReplicaSetReconciler) error {
```

Comment:
Same as above

Reply:
I updated this comment to give a bit more info on what it's suppose to do.